c/public/lib/source/cffwrite/cffwrite.c

```diff
@@ -1493,10 +1493,19 @@ int cfwBegFont(cfwCtx g, cfwMapCallback *map, unsigned long maxNumSubrs) {
 static void orderCIDKeyedGlyphs(controlCtx h) {
     cfwCtx g = h->g;
     long i;
-    unsigned char fdmap[256];
+    uint16_t *fdmap;
+    size_t fdmap_size;
     long nGlyphs = h->_new->glyphs.cnt;
     Glyph *glyphs = h->_new->glyphs.array;
 
+    fdmap_size = h->_new->FDArray.cnt * sizeof(uint16_t);
+    fdmap = malloc(fdmap_size);
```

Comment:
Instead of calling raw `malloc` directly, which may leak if `cfwErrNoCID0` error condition is hit below, let's allocate as a dynamic array by using `h->g->dnaSafe`. Still may leak, but it allows the client to choose to sub-allocate memory for AFDKO and free entirely at the end.

Reply:
Done. Thanks.